src/ResourceManager/DevSpaces/Commands.DevSpaces/Commands/UpdateAzureRmDevSpacesController.cs

```diff
@@ -0,0 +1,125 @@
+using System;
+using System.Collections;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.DevSpaces.Models;
+using Microsoft.Azure.Commands.DevSpaces.Properties;
+using Microsoft.Azure.Commands.DevSpaces.Utils;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Commands.ResourceManager.Common.Tags;
+using Microsoft.Azure.Management.DevSpaces;
+using Microsoft.Azure.Management.DevSpaces.Models;
+
+namespace Microsoft.Azure.Commands.DevSpaces.Commands
+{
+    [Cmdlet(VerbsData.Update, DevSpacesControllerNoun, DefaultParameterSetName = DevSpacesControllerNameParameterSet, SupportsShouldProcess = true)]
+    [OutputType(typeof(PSController))]
+    public class UpdateAzureRmDevSpacesController : DevSpacesCmdletBase
+    {
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ParameterSetName = DevSpacesControllerNameParameterSet,
+            HelpMessage = "Resource group name")]
+        [ResourceGroupCompleter()]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ParameterSetName = DevSpacesControllerNameParameterSet,
+            HelpMessage = "DevSpaces controller name.")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = ResourceIdParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The DevSpaces resource id")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(
+            ParameterSetName = InputObjectParameterSet,
+            Mandatory = true,
+            ValueFromPipeline = true)]
+        [ValidateNotNullOrEmpty]
+        public PSController InputObject { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "A hash table which represents resource tags.")]
+        public Hashtable Tag { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            base.ExecuteCmdlet();
+            var msg = $"{Name} in {ResourceGroupName}";
+            if (ShouldProcess(msg, Resources.UpdatingADevSpacesController))
+            {
+                RunCmdLet(UpdateDevSpacesControllerAction);
+            }
+        }
+
+        private void UpdateDevSpacesControllerAction()
+        {
+            switch (ParameterSetName)
+            {
+                case DevSpacesControllerNameParameterSet:
+                    UpdateDevSpacesController();
+                    break;
+
+                case ResourceIdParameterSet:
+                    string resourceGroup, name;
+                    if (!ConversionUtils.TryParseResourceId(ResourceId, ConversionUtils.DevSpacesControllerResourceTypeName, out resourceGroup, out name))
```

Comment:
same comment here.

Reply:
ConversionUtils.TryParseResourceId is implemented on top of ResourceIdentifier to make sure that we are parsing DevSpace or AKS ID.